### PR TITLE
New Serialization Proto

### PR DESCRIPTION
**Pull request by houseroad**

### CODE REVIEW

File: caffe2/proto/torch.proto [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+
+  repeated Dimension stride = 51;
+}
+
+// Types
+//
+// The standard ONNX data types.
+message TypeProto {
+
+  message Tensor {
```
Comment (houseroad) [inline caffe2/proto/torch.proto]: Maybe we can have new names for these types.

**Comment by li-roy** [comment]
This looks good in terms of being able to serialize the pytorch IR.

Are there any other considerations to make? @zdevito

File: caffe2/proto/torch.proto [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+  // stored inside. Refer to https://github.com/onnx/onnx/blob/master/docs/TypeDenotation.md#type-denotation-definition
+  // for pre-defined type denotations.
+  optional string denotation = 6;
+}
+
+// TensorProtos stores multiple TensorProto objects in one single proto. This
+// is useful for small tensors; For anything big, consider using a DB for
+// storage.
+message TensorProtos {
+  repeated TensorProto protos = 1;
```
Comment (li-roy) [inline caffe2/proto/torch.proto]: Just wondering, what's the usecase for this?

Comment (houseroad) [reply caffe2/proto/torch.proto]: It's for Caffe2, they are used at several places, for example:
https://github.com/pytorch/pytorch/blob/cca247635c6edb323176eeac7a18d3e9ab71c558/caffe2/operators/tensor_protos_db_input.cc

**Review by xiaomengy — APPROVED** [review]

File: caffe2/proto/torch.proto [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+  // optional string domain = 9;
+
+  // Graph type, equivalent to Caffe2's NetDef.type
+  optional string type = 51;
+}
+
+// Tensors
+//
+// A serialized tensor value.
+message TensorProto {
```
Comment (dzhulgakov) [inline caffe2/proto/torch.proto]: it's good for review - but overall - if the proto is the same as ONNX - shall we just depend on onnx protobuf? or there are annoyances with build in this case?

File: caffe2/proto/torch.proto [hunk 4]
```diff
@@ -0,0 +1,599 @@
...
+  // DO NOT USE the following fields, they were deprecated from earlier versions.
+  // repeated string input = 3;
+  // repeated string output = 4;
+  // optional int64 ir_version = 6;
+  // optional int64 producer_version = 7;
+  // optional string producer_tag = 8;
+  // optional string domain = 9;
+
+  // Graph type, equivalent to Caffe2's NetDef.type
+  optional string type = 51;
```
Comment (dzhulgakov) [inline caffe2/proto/torch.proto]: so NetDef type is more of an executor annotation (simple/dag/etc). It's probably better to encode it as an annotation

File: caffe2/proto/torch.proto (hunk 3 above)
Comment (houseroad) [reply caffe2/proto/torch.proto]: After a thorough thinking, I prefer not referring to onnx proto for code size purpose.

**Pull request closed without merging**